feature-libs/checkout/components/order-confirmation/components/order-confirmation-items/order-confirmation-items.component.ts

```diff
@@ -19,16 +19,13 @@ export class OrderConfirmationItemsComponent implements OnInit, OnDestroy {
   order$: Observable<Order>;
   orderPromotions$: Observable<PromotionResult[]>;
 
-  constructor(
-    protected checkoutService: CheckoutFacade,
-    protected promotionService: PromotionService
-  ) {}
+  constructor(protected checkoutService: CheckoutFacade) {}
 
   ngOnInit() {
     this.order$ = this.checkoutService.getOrderDetails();
-    this.orderPromotions$ = this.promotionService.getOrderPromotions(
```

Comment:
we can remove the `this.orderPromotions$`, and in html, get `appliedOrderPromotions` directly from order. 
replace 
```
   <ng-container *ngIf="orderPromotions$ | async as orderPromotions">
      <cx-promotions [promotions]="orderPromotions"></cx-promotions>
   </ng-container>
```
to `<cx-promotions [promotions]="order.appliedOrderPrmotions ?? []"></cx-promotions>`

But, this change is not wrong. So, it is up to you.